<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SceneGraphFrame</class>
 <widget class="QWidget" name="SceneGraphFrame">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>331</width>
    <height>529</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <property name="whatsThis">
   <string>The scene graph frame displays and grants access to the scene graph of a task.
It is separated in a header area, where you can select the task to show the scene graph from, the scene graph area in the center, where you can explore the single nodes of the graph, and the property area at the bottom, where you can inspect and change properties of selected nodes in the scene graph.</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <property name="margin">
    <number>0</number>
   </property>
   <item>
    <widget class="QWidget" name="headerWidget" native="true">
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="leftMargin">
       <number>6</number>
      </property>
      <property name="topMargin">
       <number>6</number>
      </property>
      <property name="rightMargin">
       <number>6</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="headerLayout">
        <item>
         <widget class="QLabel" name="taskLabel">
          <property name="text">
           <string>Task:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="taskComboBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="whatsThis">
           <string>Select the task to display a scene graph from.
Currently only spark server thread scene graphs can be displayed.</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QWidget" name="toolButtonWidget" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <property name="spacing">
          <number>4</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QToolButton" name="updateButton">
           <property name="toolTip">
            <string>Update all</string>
           </property>
           <property name="whatsThis">
            <string>Update the whole scene graph.</string>
           </property>
           <property name="text">
            <string>Update</string>
           </property>
           <property name="icon">
            <iconset resource="../../imageresources.qrc">
             <normaloff>:/arrow_refresh</normaloff>:/arrow_refresh</iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="expandButton">
           <property name="toolTip">
            <string>Expand all</string>
           </property>
           <property name="whatsThis">
            <string>Expands all scene graph nodes.</string>
           </property>
           <property name="text">
            <string>Expand All</string>
           </property>
           <property name="icon">
            <iconset resource="../../imageresources.qrc">
             <normaloff>:/arrow_expand</normaloff>:/arrow_expand</iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="collapseButton">
           <property name="toolTip">
            <string>Collapse all</string>
           </property>
           <property name="whatsThis">
            <string>Collapses all scene graph nodes.</string>
           </property>
           <property name="text">
            <string>Collapse All</string>
           </property>
           <property name="icon">
            <iconset resource="../../imageresources.qrc">
             <normaloff>:/arrow_collapse</normaloff>:/arrow_collapse</iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="updateCheckBox">
           <property name="toolTip">
            <string>Automatic update On/Off</string>
           </property>
           <property name="whatsThis">
            <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;Activates or deactivates automatic updating of the whole scene graph.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset resource="../../imageresources.qrc">
             <normaloff>:/alarm</normaloff>:/alarm</iconset>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>189</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="contentWidget" native="true">
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QSplitter" name="contentSplitter">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <widget class="QStackedWidget" name="stackedWidget">
         <property name="lineWidth">
          <number>0</number>
         </property>
         <widget class="QWidget" name="page"/>
         <widget class="QWidget" name="page_2"/>
        </widget>
        <widget class="SceneGraphFrameUtil::PropertyFrame" name="propertyFrame">
         <property name="whatsThis">
          <string/>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Plain</enum>
         </property>
         <property name="lineWidth">
          <number>0</number>
         </property>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SceneGraphFrameUtil::PropertyFrame</class>
   <extends>QFrame</extends>
   <header>scenegraphframe_propertyframe.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../imageresources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>expandButton</sender>
   <signal>clicked()</signal>
   <receiver>SceneGraphFrame</receiver>
   <slot>expandAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>44</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>36</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>collapseButton</sender>
   <signal>clicked()</signal>
   <receiver>SceneGraphFrame</receiver>
   <slot>collapseAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>63</x>
     <y>43</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>66</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>updateButton</sender>
   <signal>clicked()</signal>
   <receiver>SceneGraphFrame</receiver>
   <slot>updateData()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>49</y>
    </hint>
    <hint type="destinationlabel">
     <x>-6</x>
     <y>27</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>updateCheckBox</sender>
   <signal>clicked(bool)</signal>
   <receiver>SceneGraphFrame</receiver>
   <slot>toggleTimer(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>93</x>
     <y>46</y>
    </hint>
    <hint type="destinationlabel">
     <x>334</x>
     <y>38</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>taskComboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>SceneGraphFrame</receiver>
   <slot>chooseTask(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>300</x>
     <y>23</y>
    </hint>
    <hint type="destinationlabel">
     <x>333</x>
     <y>11</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>chooseTask(int)</slot>
  <slot>collapseAll()</slot>
  <slot>expandAll()</slot>
  <slot>updateData()</slot>
  <slot>toggleTimer(bool)</slot>
 </slots>
</ui>
